### DISCUSSION TITLE
Public key to puzzle Hash conversion

### DISCUSSION THREAD

**Discussion by itorry** [Chia-Network/chia-blockchain#7782] [Support]
I want to generate the address of the main network now. It should be possible to convert the address from the public key to the puzzle hash. At present, there are still problems in the conversion from the public key to the puzzle hash. I'm not a python developer and I'm not familiar with the python language. It's hard to read the source code. Does anyone know any other way to get the puzzle hash from the public key, such as RPC, or the Java language or JS language,thanks

**Comment by keliew** [comment]
This? https://www.chiaexplorer.com/tools/address-puzzlehash-converter

**Reply by itorry** [reply]
Thanks, but this isn't what I need,This is the puzzle hash to address, I want is to convert from the public key to the puzzle hash
